docs/content/docs/start/react-tutorial.md

```diff
@@ -0,0 +1,295 @@
+    - `updateLocalTimestamp` is called immediately to ensure that `localTimestamp` is initialized with the current shared timestamp value.
+
+    ```js
+    const { sharedTimestamp } = fluidSharedMap;
+    const updateLocalTimestamp = () => setLocalTimestamp({ time: sharedTimestamp.get("time") });
+    
+    updateLocalTimestamp();
+    ```
+
+1. To ensure that the `localTimestamp` state is updated whenever the `fluidSharedMap` is changed _even by other clients_, replace `TODO 5` with the following code. Note that because `updateLocalTimestamp` calls the state-setting function `setTimestamp`, a rerender is triggered whenever any client changes the Fluid `fluidSharedMap`.
+
+    ```js
+    sharedTimestamp.on("valueChanged", updateLocalTimestamp);
+   ```
+
+1. It is a good practice to deregister event handlers when the React component dismounts, so replace `TODO 6` with the following code.
+
+    ```js
+    return () => { sharedTimestamp.off("valueChanged", updateLocalTimestamp) }
+   ```
+
+### Create the view
+
+Below the `useEffect` hook, replace the `return ();` line with the following code. Note about this code:
+
+- If the `localTimestamp` state has not been initialized, a blank screen is rendered.
+- The `sharedTimestamp.set` method sets the _key_ of the `fluidSharedMap` object to "time" and the _value_ to the current UNIX epoch time. This triggers the `valueChanged` event on the object, so the `updateLocalTimestamp` function runs and sets the `localTimestamp` state to the same object; for example, `{time: "1615996266675"}`. The `App` component rerenders and the `<span>` is updated with the latest timestamp.
+- All other clients update too because the Fluid server propagates the change to the `fluidSharedMap` on all of them and this `valueChanged` event updates the `localTimestamp` state on all of them.
+
+```js
+if (localTimestamp) {
+    return (
+        <div className="App">
+            <button onClick={() => fluidSharedMap.sharedTimestamp.set("time", Date.now().toString())}>
+                Get Time
+            </button>
+            <span>{localTimestamp.time}</span>
+        </div>
+    )
+} else {
+     return <div/>;
+}
+```
+
+## Start the Fluid server and run the application
+
+In the Command Prompt, run the following command to start the FRS server. Note that `tinylicious` is the name of the FRS that runs on localhost.
+
+```dotnetcli
+npx tinylicious
+```
+
+Open a new Command Prompt and navigate to the root of the project; for example, `C:/My Fluid Projects/fluid-react-tutorial`. Start the application server with the following command. The application opens in your browser. This may take a few minutes.
+
+```dotnetcli
+npm run start
+```
+
+{{< callout note >}}
+
+If you are prompted to run the application server on another port, press Enter to accept.
+
+{{< /callout >}}
```

Comment:
I think this will cause problems later if it happens, because the app assumes the default port.

Reply:
```suggestion
```